app/services/foreman/plugin.rb

```diff
@@ -275,5 +276,11 @@ def apipie_ignored_controllers(controllers = nil)
     def register_custom_status(klass)
       HostStatus.status_registry.add(klass)
     end
+
+    # register a provision method
+    def provision_method(name, friendly_name)
+      return if @provision_methods.key?(name)
+      @provision_methods[name] = friendly_name
```

Comment:
It might be best to convert `name` to `name.to_s` on entry to this method, I instinctively wrote `provision_method :test, ..` and it threw a runtime error later when viewing the New Host form.